### ISSUE TITLE
Support USB powering (warn/omit buck and supervisor)

### ISSUE THREAD

**Issue by mightymos** [mulcmu/esphome-ge-laundry-uart#9]
Success!

My onion board was connected by Ethernet cable only to GTW460ASJ6WW (washer).
However by monitoring USB serial output ESP32 would continuously reset when attempting wifi setup.
This implied the washer +9V supply was not handling a current increase during wifi setup.

I decided to try powering from USB wall charger (had in drawer 5V 700mA capacity).
I needed to remove buck converter since it appeared to be back feeding.
I also had to remove APX803L since it was holding ESP32 in reset.

Finally connected both Ethernet cable and USB charger to ESP32 USB port and powered up correctly!
Added the esphome integration in Home Assistant and shows sensors as Unknown state, but they are timestamped as updating.

I think the supervisor is APX803L40-50SA for 5V threshold.
I think USB power results in 4.85 V so supervisory is holding reset.
![washer_onion_hookup](https://user-images.githubusercontent.com/26175234/177259467-5fbb381b-e9f2-40ef-8198-a3a56c427aeb.jpg)

If desired to use USB power could 1) add a warning not to populate or 2) use lower threshold supervisor?

[esp32_washer_powered_reset.txt](https://github.com/mulcmu/esphome-ge-laundry-uart/files/9043614/esp32_washer_powered_reset.txt)
[GTW460ASJ6WW_Washer_WiringDiagram.pdf](https://github.com/mulcmu/esphome-ge-laundry-uart/files/9043615/GTW460ASJ6WW_Washer_WiringDiagram.pdf)

**Comment by mulcmu** [comment]
The DC/DC converter comes with an adjustment pot and also solder set points for a fixed voltage.  I removed the adjustment pot and (in lieu of cutting the pot trace as recommended by supplier) and soldered the 5V jumper closed.  This gave right around 5V under load with a 15V input on the bench.  I was concerned about inadvertent adjustment of the pot during use.  I wonder if the 9V supply from the washer would work if converter was adjusted in circuit to give 5V with the 9V input.

I ran into a problem with the 5V supervisor on another project and switched to 3.3V for that design.  It was an ESP8266 bare module vs ESP32 dev board.  I think not populating might be better solution as the dev board might already have an RC delay on the EN pin.

**Comment by mightymos** [comment]
For the pot if that's how it comes, would probably be better to set and then hot glue or nail polish to fix shaft non-destructively.
Maybe some ESP32 have brown out detector but it's not clear to me which from quick searches.
I think it used to be pretty common that almost all microcontrollers basically had to have power up reset built in.

With the external supervisor they sell thresholds fixed at increments of tenth of a volt.
If still wanted, could probably install one with a fixed threshold a little less than 5V to give some wiggle room (e.g., 4.8 V).
Specific threshold chosen depends on what the ESP32 requires as minimum or typical range near 5V to be considered stable.

I ended up installing SIP connector to act as socket for buck converter.
I destroyed the buck removing it. :(
I can try some experiments, what's the part number on that buck board?

**Comment by mulcmu** [comment]
Here was the [link](https://www.google.com/url?sa=t&source=web&rct=j&url=https://www.amazon.com/Regulator-Module-Voltage-Reducer-Adjustable/dp/B08R6337QY&ved=2ahUKEwie5u7qjob5AhX2FFkFHTfaAIEQFnoECBgQAQ&usg=AOvVaw21JQGmN4V32SUN4N8p0bzJ) for buck converter B08R6337QY

**Comment by mightymos** [comment]
I did buy a slightly different buck module (B0758ZTS61 - just to receive fewer quantity).
I set it about 5V on breadboard by potentiometer and confirmed it maintained voltage in circuit with no further adjustment.
I think there's an OR-ing diode in there too (in case USB power is also connected) but apparently not much voltage drop.

The board has been powered up for several days with no drops from wifi or Home Assistant.
No voltage supervisor has been reinstalled so can not speak to that.

Unfortunately have no apparent communication to washer but confirms working hardware/software.
